test/common/router/config_impl_speed_test.cc

```diff
@@ -132,13 +142,260 @@ static void bmRouteTableSizeWithExactPathMatch(benchmark::State& state) {
+    ->ArgPair(5000, 1000)
+    ->ArgPair(10000, 1000)
+    ->ArgPair(1, 2000)
+    ->ArgPair(10, 2000)
+    ->ArgPair(50, 2000)
+    ->ArgPair(100, 2000)
+    ->ArgPair(500, 2000)
+    ->ArgPair(1000, 2000)
+    ->ArgPair(1500, 2000)
+    ->ArgPair(2000, 2000)
+    ->ArgPair(5000, 2000)
+    ->ArgPair(10000, 2000);
+
+/**
+ * Benchmark a route table with exact path matchers with different path rule length:
+ * - /abcdefghij...<repeat n times>_n
+ * - etc.
+ */
+static void bmRouteTableSizeWithExactPathMatchVaryPathLen(benchmark::State& state) {
+  std::string path_rule = GetPrefixPathRuleFromBase(state.range(1));
+  bmRouteTableSize(state, RouteMatch::PathSpecifierCase::kPath,
+                  path_rule, path_rule);
+}
+
+// Arg1 is the path rule table size. Arg2 is the path rule length.
+BENCHMARK(bmRouteTableSizeWithExactPathMatchVaryPathLen)
+    ->ArgPair(1, 10)
+    ->ArgPair(10, 10)
+    ->ArgPair(50, 10)
+    ->ArgPair(100, 10)
+    ->ArgPair(500, 10)
+    ->ArgPair(1000, 10)
+    ->ArgPair(1500, 10)
+    ->ArgPair(2000, 10)
+    ->ArgPair(5000, 10)
+    ->ArgPair(10000, 10)
+    ->ArgPair(1, 50)
+    ->ArgPair(10, 50)
+    ->ArgPair(50, 50)
+    ->ArgPair(100, 50)
+    ->ArgPair(500, 50)
+    ->ArgPair(1000, 50)
+    ->ArgPair(1500, 50)
+    ->ArgPair(2000, 50)
+    ->ArgPair(5000, 50)
+    ->ArgPair(10000, 50)
+    ->ArgPair(1, 100)
+    ->ArgPair(10, 100)
+    ->ArgPair(50, 100)
+    ->ArgPair(100, 100)
+    ->ArgPair(500, 100)
+    ->ArgPair(1000, 100)
+    ->ArgPair(1500, 100)
+    ->ArgPair(2000, 100)
+    ->ArgPair(5000, 100)
+    ->ArgPair(10000, 100)
+    ->ArgPair(1, 500)
+    ->ArgPair(10, 500)
+    ->ArgPair(50, 500)
+    ->ArgPair(100, 500)
+    ->ArgPair(500, 500)
+    ->ArgPair(1000, 500)
+    ->ArgPair(1500, 500)
+    ->ArgPair(2000, 500)
+    ->ArgPair(5000, 500)
+    ->ArgPair(10000, 500)
+    ->ArgPair(1, 1000)
+    ->ArgPair(10, 1000)
+    ->ArgPair(50, 1000)
+    ->ArgPair(100, 1000)
+    ->ArgPair(500, 1000)
+    ->ArgPair(1000, 1000)
+    ->ArgPair(1500, 1000)
+    ->ArgPair(2000, 1000)
+    ->ArgPair(5000, 1000)
+    ->ArgPair(10000, 1000)
+    ->ArgPair(1, 2000)
+    ->ArgPair(10, 2000)
+    ->ArgPair(50, 2000)
+    ->ArgPair(100, 2000)
+    ->ArgPair(500, 2000)
+    ->ArgPair(1000, 2000)
+    ->ArgPair(1500, 2000)
+    ->ArgPair(2000, 2000)
+    ->ArgPair(5000, 2000)
+    ->ArgPair(10000, 2000);
+
+/**
+ * Benchmark a route table with regex path matchers in the form of:
+ * - "^/api/_?/v\\d{2}"                                       : ~10 bytes regex complexity
+ * - "^/videos/_?/v\\d{3}/ba[r|z].m?[4s|pg|p4]"               " ~50 bytes regex complexity
+ * - "^/.* /foo\\d{3}\\W{1,2}V+T?/ba[r|z]/A{2}.[h|w][^a-z]{1}": ~100 bytes regex complexity
+ * - etc.
+ */
+
+// Generate RegEx path rule based on path rule length.
+static std::string GetRegexPathRuleFromBase(int path_len) {
+  std::string path_rule = "";
+
+  switch (path_len) {
+    case 10:
+      absl::StrAppend(&path_rule, "^/api/_?/v\\d{2}");
```

Comment:
This is kind of a weird way to structure things: why are we going from magic numbers to strings? Should we be accepting anyh number and repeatedly building, or use enums, which are more semantically meaningful?